runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -380,10 +380,9 @@ NNFW_STATUS nnfw_session::apply_tensorinfo(uint32_t index, nnfw_tensorinfo ti)
 {
   // sanity check
   {
-    if (!primary_subgraph() || primary_subgraph()->isBuildingPhase())
+    if (!(isStateModelLoaded() || isStatePrepared()))
```

Comment:
Would `if (!(isStateModelLoaded())` be enough?

Reply:
Actually there is no way to get in the state of BUILDING_PHASE for a graph through NNFWAPI. However I added assertions to state checking functions.